test/correctness/bitwise_ops.cpp

```diff
@@ -129,22 +129,142 @@ int main(int argc, char **argv) {
         shift_amount = std::min(31, shift_amount);
         int correct = lhs >> shift_amount;
         if (im9(x) != correct) {
-            printf("%x >> (uint8_t)%x -> %x instead of %x\n",
-                   input(x), input(x+1), im9(x), correct);
+            printf("%d >> %d -> %d instead of %d\n",
+                   input(x), shift_amount, im9(x), correct);
             return -1;
         }
     }
 
-    // bitwise and on mixed types
+    // arithmetic shift left with signed type (positive value)
     Func f10;
+    Expr a10 = reinterpret<int>(input(x));
+    Expr b10 = reinterpret<int>(input(x+1));
+    f10(x) = a10 << (b10 & 0xf);
```

Comment:
Why not 31 here, instead of 15? (for 32-bit integers).

Reply:
Yup, it should be 31. Changed it to that.